modules/db/src/main/java/org/mule/module/db/internal/config/domain/database/DbConfigDefinitionParser.java

```diff
@@ -123,9 +123,9 @@ else if (id == Types.STRUCT)
                         customDbTypes.add(new StructDbType(id, name));
                     }
                 }
-                else if (id == Types.CLOB)
+                else if (types.contains(new ResolvedDbType(id, name)))
                 {
-                    customDbTypes.add(new ClobResolvedDataType(id, name));
+                    customDbTypes.add(types.get(types.indexOf(new ResolvedDbType(id, name))));
```

Comment:
This was done in order to always retrieve the subclass of ResolvedDataType which is declared in JdbcDataType. When you use 

types.get(types.indexOf(new ResolvedDbType(id, name))) 

as the equals is overriden and it only takes into account the id and the name, it will get ClobResolvedDataType or BlobResolvedDataType according to the id and name attributes of the ResolvedDbType used to get the class in JdbcTypes.